eclipsing-binary-simulator/src/BinarySystemView.jsx

```diff
@@ -219,21 +241,15 @@ export default class BinarySystemView extends React.Component {
         if (initObject.phi!=undefined && !(initObject.phi<-90 || initObject.phi>90)) this._phi = initObject.phi*(Math.PI/180);
         if (initObject.theta!=undefined) this._theta = initObject.theta*(Math.PI/180);
         if (initObject.scale!=undefined) this._scale = initObject.scale;
-        if (initObject.phase!=undefined) this._phase = (initObject.phase%1 + 1)%1;
-        if (initObject.radius1!=undefined) this._radius1 = initObject.radius1;
-        if (initObject.radius2!=undefined) this._radius2 = initObject.radius2;
-        if (initObject.eccentricity!=undefined) this._eccentricity = initObject.eccentricity;
-        if (initObject.mass1!=undefined) this._mass1 = initObject.mass1;
-        if (initObject.mass2!=undefined) this._mass2 = initObject.mass2;
-        if (initObject.separation!=undefined) this._separation = initObject.separation;
         if (initObject.linePhi!=undefined) this._linePhi = initObject.linePhi;
         if (initObject.lineTheta!=undefined) this._lineTheta = initObject.lineTheta;
         if (initObject.showLine!=undefined) this._showLine = Boolean(initObject.showLine);
         if (initObject.lineExtra!=undefined) this._lineExtra = initObject.lineExtra;
 
-        this._massTotal = this._mass1 + this._mass2;
-        this._a1 = this._separation*this._mass2/this._massTotal;
-        this._a2 = this._separation*this._mass1/this._massTotal;
+        this._massTotal = this.props.star1Mass + this.props.star2Mass;
+
+        this._a1 = this.props.separation * this.props.star2Mass / this._massTotal;
```

Comment:
Noting the `star2Mass` is used in the `a1` variable, and `star1Mass` is used in the `a2` variable. Likely fine, but worth a glance.

Reply:
Yes, this is correct, according to the original code. Thanks for confirming.